src/Middleware/ResponseCaching/src/Streams/ResponseCachingStream.cs

```diff
@@ -145,13 +148,13 @@ public override async Task WriteAsync(byte[] buffer, int offset, int count, Canc
 
             if (BufferingEnabled)
             {
-                if (_segmentWriteStream.Length + count > _maxBufferSize)
```

Comment:
Another place we ignored offset before

Reply:
There was a validation check missing for offset, but count/length is the important part of this check.